packages/http/libraries/core/src/adapter/InversifyHttpAdapter.ts

```diff
@@ -0,0 +1,209 @@
+import { Container } from '@inversifyjs/container';
+import { getReflectMetadata } from '@inversifyjs/reflect-metadata-utils';
+
+import { ControllerMetadata } from '../models/ControllerMetadata';
+import { ControllerMethodMetadata } from '../models/ControllerMethodMetadata';
+import { ControllerMethodParameterMetadata } from '../models/ControllerMethodParameterMetadata';
+import { METADATA_KEY } from '../models/MetadataKey';
+import { RequestMethodParameterType } from '../models/RequestMethodParameterType';
+import { RouterParams } from '../models/RouterParams';
+
+export abstract class InversifyHttpAdapter<
+  TRequest,
+  TResponse,
+  TNextFunction extends (err?: unknown) => void,
+> {
+  readonly #container: Container;
+
+  constructor(container: Container) {
+    this.#container = container;
+  }
+
+  protected _buildServer(): void {
+    this.#registerControllers();
+  }
+
+  #registerControllers(): void {
+    const controllerMetadataList: ControllerMetadata[] | undefined =
+      getReflectMetadata(Reflect, METADATA_KEY.controller);
+
+    if (controllerMetadataList === undefined) {
+      throw new Error('No controllers found');
+    }
```

Comment:
I am not sure we should throw here. if we do, let's have custom errors the same way the container core library has :)

Reply:
I refactored this error in that way